### PR TITLE
add test for internal/errors/runtime

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

I added the test for `internal/errors/runtime.go` and add comment for each function.

### Environment:

- Go Version: 1.15.7
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.13.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/errors/runtime.go [hunk 1]
```diff
@@ -20,16 +20,17 @@ package errors
 import "runtime"
 
 var (
-	// Runtime.
-
+	// ErrPanicRecovered represents a function to generate an error that panic recovered.
```
Comment (hlts2) [inline internal/errors/runtime.go:23]:
```suggestion
	// ErrPanicRecovered represents a function to generate an error that the panic recovered.
```

File: internal/errors/runtime.go [hunk 2]
```diff
@@ -20,16 +20,17 @@ package errors
...
 	ErrPanicRecovered = func(err error, rec interface{}) error {
 		return Wrap(err, Errorf("panic recovered: %v", rec).Error())
 	}
 
+	// ErrPanicString represents a function to generate an error that panic recovered with string message.
```
Comment (hlts2) [inline internal/errors/runtime.go:28]:
```suggestion
	// ErrPanicString represents a function to generate an error that the panic recovered with a string message.
```

File: internal/errors/runtime.go [hunk 3]
```diff
@@ -20,16 +20,17 @@ package errors
...
 	ErrPanicString = func(err error, msg string) error {
 		return Wrap(err, Errorf("panic recovered: %v", msg).Error())
 	}
 
+	// ErrRuntimeError represents a function to generate an error that panic caused by runtime error.
```
Comment (hlts2) [inline internal/errors/runtime.go:33]:
```suggestion
	// ErrRuntimeError represents a function to generate an error that the panic caused by runtime error.
```

File: internal/errors/runtime_test.go [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+		func() test {
+			r := math.MaxFloat64
+			return test{
+				name: "return an error when err is not empty and rec is int value",
+				args: args{
+					err: defaultErr,
+					rec: r,
+				},
+				want: want{
+					want: Wrap(defaultErr, Errorf("panic recovered: %v", r).Error()),
```
Comment (hlts2) [inline internal/errors/runtime_test.go:63]: Should we use New function insted of Wrap function.
because this want is equal to the test target function like this.
https://github.com/vdaas/vald/blob/1b6993be91fc53f20a143bc650e546686f97394c/internal/errors/runtime.go#L25

File: internal/errors/runtime_test.go [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	defaultErr := New("runtime panic string error")
+	defaultRuntimeErr := &runtimeErr{
+		err: New("runtime error is occured"),
```
Comment (vankichi) [inline internal/errors/runtime_test.go:357]:
```suggestion
		err: New("runtime error is occurred"),
```

File: internal/errors/runtime_test.go (hunk 4 above)
Comment (vankichi) [reply internal/errors/runtime_test.go:63]: if we'd like to test the `Wrap` function, I agree with you.
IMO, about this case, it is not needed.
How do you think?

Comment (hlts2) [reply internal/errors/runtime_test.go:63]: I got it 👍 I agree with you!

**Review by hlts2 — APPROVED** [review]

File: internal/errors/runtime_test.go [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+				},
+				want: want{
+					want: Wrap(defaultErr, Errorf("panic recovered: %v", r).Error()),
+				},
+			}
+		}(),
+		func() test {
+			r := &defaultErr
+			return test{
+				name: "return an error when err is not empty and rec is boolean",
```
Comment (kevindiu) [inline internal/errors/runtime_test.go:171]:
```suggestion
				name: "return an error when err is not empty and rec is pointer",
```

File: internal/errors/runtime_test.go [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+			return test{
+				name: "return an error when err not nil and msg is not empty",
+				args: args{
+					msg: defaultMsg,
+				},
+				want: want{
+					want: Wrap(nil, Errorf("panic recovered: %v", defaultMsg).Error()),
+				},
+			}
+		}(),
```
Comment (kevindiu) [inline internal/errors/runtime_test.go:296]: do we need a test for both `err` and `msg` is nil?

File: internal/errors/runtime_test.go [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+			return test{
+				name: "return an error when err is nil and msg is not empty",
+				args: args{
+					r: defaultRuntimeErr,
+				},
+				want: want{
+					want: Wrap(nil, Errorf("system panicked caused by runtime error: %v", defaultRuntimeErr).Error()),
+				},
+			}
+		}(),
```
Comment (kevindiu) [inline internal/errors/runtime_test.go:393]: same, do we need a test for both `err` and `msg` is nil?

File: internal/errors/runtime_test.go [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+
+			got := ErrPanicString(test.args.err, test.args.msg)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+type runtimeError runtime.Error
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/errors/runtime_test.go:321]: maybe you can remove it?

**Review by kevindiu — CHANGES_REQUESTED** [review]

**Review by kevindiu — APPROVED** [review]
LGTM
